src/components/SettingsInputText/SettingsInputText.vue

```diff
@@ -0,0 +1,162 @@
+<!--
+  - @copyright Copyright (c) 2019 Julius Härtl <jus@bitgrid.net>
+  -
+  - @author Julius Härtl <jus@bitgrid.net>
+  - @author Greta Doci <gretadoci@gmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<form @submit.prevent="">
+		<div class="input-wrapper">
+			<label :for="id">{{ label }}</label>
+			<input :id="id" type="text" :value="inputVal"
+				:disabled="disabled" @input="$emit('input', $event.target.value)">
+			<input type="submit" class="icon-confirm" value=""
+				@click="$emit('update', inputVal)">
+		</div>
+		<p v-if="hint !== ''" class="hint">
+			{{ hint }}
+		</p>
+	</form>
+</template>
+
+<script>
+import GenRandomId from 'src/utils/GenRandomId'
+export default {
+	name: 'SettingsInputText',
+	props: {
+		/**
+		 * label of the select group element
+		 */
+		label: {
+			type: String,
+			required: true
+		},
+
+		/**
+		* hint of the select group input
+		*/
+		hint: {
+			type: String,
+			default: ''
+		},
+
+		/**
+		* value of the select group input
+		 */
+		value: {
+			type: String,
+			default: ''
+		},
+
+		/**
+		 * disabled state of the settings select group input
+		 */
+		disabled: {
+			type: Boolean,
+			default: false
+		},
+
+		/**
+		 * id attribute of the select group element
+		 */
+		id: {
+			type: String,
+			default: () => 'action-' + GenRandomId(),
+			validator: id => id.trim() !== ''
+		}
+	},
+	data() {
+		return {
+			inputVal: this.value
+		}
+	},
+	computed: {
+		/**
+		* @returns {string}
+		* @constructor
+		*/
+		GenRandomId() {
+			return 'settings-input-text-' + this.GenRandomId
+		}
+	},
+	watch: {
+		value(newVal) {
+			this.inputVal = this.value
```

Comment:
Why keeping a local data here?
The input value should be enough, I don't thing there is a need to keep track of this here with a watcher :thinking: